src/app/data/services/data_service.ts

```diff
@@ -30,6 +32,15 @@ class DataService {
     return this.loadContent(contentBlock, true);
   }
 
+  loadReadonlyContentFromCache(contentBlock:ContentBlock):ContentBlock {
+    const cacheKey = this.cacheKey(contentBlock, true);
+    const hash = this.localStorageService.fetch(cacheKey);
+
+    if(!hash) return contentBlock;
+
+    return this.dataConverter.extractObjectFromHash(contentBlock, hash);
```

Comment:
on second thought: no.
Everything I try makes it more unreadable. The way it is suggests that we early return the default content block on a cache miss. We want the programmer to read it like that.